### PR TITLE
Feat/remove legacy metrics

### PR DESCRIPTION
**Pull request by ivarconr**
Related change in frontend: https://github.com/Unleash/unleash-frontend/pull/553

### CODE REVIEW

**Comment by ivarconr** [comment]
[Remove old metrics service.](https://trello.com/c/PHHyvh9h/651-remove-old-metrics-service)

File: src/lib/routes/client-api/metrics.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export default class ClientMetricsController extends Controller {
     async registerMetrics(req: IAuthRequest, res: Response): Promise<void> {
         const { body: data, ip: clientIp, user } = req;
         data.environment = this.resolveEnvironment(user, data);
-        await this.metrics.registerClientMetrics(data, clientIp);
+        await this.clientInstanceService.registerInstance(data, clientIp);
```
Comment (chriswk) [inline src/lib/routes/client-api/metrics.ts:56]: I like it, it's more clear what data is being registered.

File: src/lib/services/client-metrics/instance-service.test.ts [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+ *    await flushPromises(); // this is required to make it work!
+ *
+ *    expect(spy).toHaveBeenCalledTimes(1);
+ *
+ *    jest.useRealTimers();
+ * });
+ */
+function flushPromises() {
+    return Promise.resolve(setImmediate);
+}
```
Comment (chriswk) [inline src/lib/services/client-metrics/instance-service.test.ts:38]: Nice, then our metrics tests should not be to blame for jest complaining about hanging timers :)

**Review by chriswk — APPROVED** [review]
A few positive observations, and not seeing anything that looks wrong, so I'd say GO GO
:+1:

**Pull request merged**